### ISSUE TITLE
Ohai detects server inside gce under specific conditions

### ISSUE THREAD

**Issue by josacar** [chef/ohai#399]
### Version:
7.2.4

### Environment
Debian Squeeze and Wheezy under EC2 cloud

### Scenario:

In a EC2 cloud, fter Ohai sets automatic attributes in a chef run, `cloud` attribute contains empty values in IP fields:

```
chef > node['cloud']
 => {"public_ips"=>[], "private_ips"=>[], "public_ipv4"=>[], "local_ipv4"=>[]}
```

Also debug cloud plugin output:

```
...
DEBUG: can_metadata_connect? == true
DEBUG: looks_like_gce? == true
...
DEBUG: has_ec2_mac? == true
DEBUG: can_metadata_connect? == true
DEBUG: looks_like_ec2? == true
DEBUG: EC2 shows unsupported metadata version: 2014-02-25
DEBUG: EC2 metadata version: 2012-01-12
DEBUG: EC2 shows unsupported metadata version: 2014-02-25
DEBUG: EC2 metadata version: 2012-01-12
...
DEBUG: Plugin CloudV2 threw #<NoMethodError: undefined method `[]' for nil:NilClass>
DEBUG: .../ohai-7.2.4/lib/ohai/plugins/cloud_v2.rb:125:in `get_gce_values'
DEBUG: .../ohai-7.2.4/lib/ohai/plugins/cloud_v2.rb:278:in `block (2 levels) in <main>'
DEBUG: .../ohai-7.2.4/lib/ohai/dsl/plugin/versionvii.rb:90:in `instance_eval'
DEBUG: .../ohai-7.2.4/lib/ohai/dsl/plugin/versionvii.rb:90:in `run_plugin'
DEBUG: .../ohai-7.2.4/lib/ohai/dsl/plugin.rb:98:in `run'
DEBUG: .../ohai-7.2.4/lib/ohai/dsl/plugin.rb:169:in `safe_run'
DEBUG: .../ohai-7.2.4/lib/ohai/runner.rb:82:in `run_v7_plugin'
DEBUG: .../ohai-7.2.4/lib/ohai/runner.rb:43:in `run_plugin'
DEBUG: .../ohai-7.2.4/lib/ohai/system.rb:96:in `block inrun_plugins'
DEBUG: .../ohai-7.2.4/lib/ohai/system.rb:95:in `each'
DEBUG: .../ohai-7.2.4/lib/ohai/system.rb:95:in `run_plugins'
DEBUG: .../ohai-7.2.4/lib/ohai/system.rb:73:in `all_plugins'
DEBUG: .../chef-11.14.6/lib/chef/client.rb:292:in `run_ohai'
DEBUG: .../chef-11.14.6/lib/chef/shell/shell_session.rb:155:in `rebuild_node'
DEBUG: .../chef-11.14.6/lib/chef/shell/shell_session.rb:55:in `block in reset!'
DEBUG: .../chef-11.14.6/lib/chef/shell/shell_session.rb:101:in `loading'
DEBUG: .../chef-11.14.6/lib/chef/shell/shell_session.rb:54:in `reset!'
DEBUG: .../chef-11.14.6/lib/chef/shell.rb:127:in `session'
DEBUG: .../chef-11.14.6/lib/chef/shell.rb:136:in `init'
DEBUG: .../chef-11.14.6/lib/chef/shell.rb:64:in `start'
DEBUG: .../chef-11.14.6/bin/chef-shell:37:in `<top (required)>'
DEBUG: .../bin/chef-shell:23:in `load'
DEBUG: .../bin/chef-shell:23:in `<main>'
DEBUG: ip_scopes: cannot load gem, plugin disabled: cannot load such file -- ipaddr_extensions
DEBUG: Plugin Cloud threw #<NoMethodError: undefined method `[]' for nil:NilClass>
DEBUG: .../gems/ohai-7.2.4/lib/ohai/plugins/cloud.rb:53:in `get_gce_values'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/plugins/cloud.rb:222:in `block (2 levels) in <main>'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/dsl/plugin/versionvii.rb:90:in `instance_eval'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/dsl/plugin/versionvii.rb:90:in `run_plugin'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/dsl/plugin.rb:98:in `run'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/dsl/plugin.rb:169:in `safe_run'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/runner.rb:82:in `run_v7_plugin'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/runner.rb:43:in `run_plugin'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/system.rb:96:in `block in run_plugins'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/system.rb:95:in `each'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/system.rb:95:in `run_plugins'
DEBUG: .../gems/ohai-7.2.4/lib/ohai/system.rb:73:in `all_plugins'
DEBUG: .../gems/chef-11.14.6/lib/chef/client.rb:292:in `run_ohai'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell/shell_session.rb:155:in `rebuild_node'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell/shell_session.rb:55:in `block in reset!'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell/shell_session.rb:101:in `loading'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell/shell_session.rb:54:in `reset!'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell.rb:127:in `session'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell.rb:136:in `init'
DEBUG: .../gems/chef-11.14.6/lib/chef/shell.rb:64:in `start'
DEBUG: .../gems/chef-11.14.6/bin/chef-shell:37:in `<top (required)>'
DEBUG: .../bin/chef-shell:23:in `load'
DEBUG: .../bin/chef-shell:23:in `<main>'...
```

This is due [gce_metadata mixin](https://github.com/opscode/ohai/blob/master/lib/ohai/mixin/gce_metadata.rb#L24) is connecting to a host without the trailing dot when trying to discover if exists or not.

### Steps to Reproduce:

- Create a resolv.conf with a `search` including a domain search like `wadus.domain.com`

- Create a wildcard DNS record `*.wadus.domain.com` pointing so a server that has port 80 opened and reachable from the instance.

- Run a chef-client or chef-shell

### Expected Result:

```
chef > node['cloud']
 => {"public_ips"=>["54.87.155.167"], "private_ips"=>["10.102.167.230"], "public_ipv4"=>"54.87.155.167", "public_hostname"=>"ec2-54-87-155-167.compute-1.amazonaws.com", "local_ip
v4"=>"10.102.167.230", "local_hostname"=>"ip-10-102-167-230.ec2.internal", "provider"=>"ec2"}
```

### Actual Result:

In a `chef-shell` ip fields are not populated:

```
chef > node['cloud']
 => {"public_ips"=>[], "private_ips"=>[], "public_ipv4"=>[], "local_ipv4"=>[]}
```

**Comment by lamont-granquist** [comment]
Yeah, PR to add the dots would be accepted.

**Comment by btm** [comment]
merged #478 to master, hopefully released shortly baring any issues.
